src/braket/experimental/autoqasm/api.py

```diff
@@ -336,7 +351,9 @@ def _convert_subroutine(
 
             # Convert the function via autograph into an oqpy subroutine
             # NOTE: Process a clone of the function so that we don't modify the original object
-            oqpy_sub = oqpy.subroutine(_wrap_for_oqpy_subroutine(_clone_function(f), options))
+            oqpy_sub = oqpy.subroutine(annotations=annotations)(
```

Comment:
Interesting interface from oqpy for adding annotations. Easy enough though

Reply:
It seems like `oqpy.subroutine(f, annotations=annotations)` should have worked, but it doesn't. @jcjaskula-aws may have more insight. I think the `oqpy.subroutine` is optimized for use as a decorator rather than being called directly. But in any case, yes, this is easy enough.